modules/core/src/v2/coins/avaxcToken.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @prettier
+ */
+import { BitGo } from '../../bitgo';
+
+import { AvaxC } from './avaxc';
+import { CoinConstructor } from '../coinFactory';
+import { coins } from '@bitgo/statics';
+
+export interface AvaxcTokenConfig {
+  name: string;
+  type: string;
+  coin: string;
+  network: string;
+  tokenContractAddress: string;
+  decimalPlaces: number;
+}
+
+export class AvaxCToken extends AvaxC {
+  public readonly tokenConfig: AvaxcTokenConfig;
+
+  constructor(bitgo: BitGo, tokenConfig: AvaxcTokenConfig) {
+    const staticsCoin = tokenConfig.network === 'Mainnet' ? coins.get('avaxc') : coins.get('tavaxc');
+    super(bitgo, staticsCoin);
+    this.tokenConfig = tokenConfig;
+  }
+
+  static createTokenConstructor(config: AvaxcTokenConfig): CoinConstructor {
+    return (bitgo: BitGo) => new AvaxCToken(bitgo, config);
```

Comment:
Can you push the env dependent constructor logic down to here? This is really what it's for (custom instance construction dependent on bitgo object).

Reply:
I can find a way. I asume that this idea is to avoid changes in registerCoinConstructor.  
The doubt is how match the tokens before registering. Do I create a temporary object for it?